tests/contract-playground/src/CostlyOperationsInsideLoops.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.4.0;
+
+// COPIED from SLITHER Wiki
+contract CostlyOperationsInLoop {
+    uint loop_count = 100;
+    uint state_variable = 0;
+
+    function bad() external {
+        for (uint i = 0; i < loop_count; i++) {
```

Comment:
@TilakMaddy , Solidity FYI, having `loop_count` in the for statement instead of a local variable incurs an SLOAD on every loop 👀

Reply:
Got it :)

Reply:
Actually, it's a good point, we need to investigate the call graph downwards even from the conidtion and the update expression of the for loop. (body is not enough) .. to find sstores. 

(In this case it was sload so I guess it was okay)

Reply:
Ah wait... this detector looks inside the entire for loop. So it should cover that case. The only false positive would be if there is a sstore in the initial expression haha .  Will resolve it as tech debt